engine/src/main/java/org/hibernate/validator/internal/util/ReflectionHelper.java

```diff
@@ -259,36 +260,27 @@ public static Object getValue(Method method, Object object) {
 	}
 
 	/**
-	 * Indicates if the type is considered indexable (ie is an {@code Iterable}, an array or a {@code Map}).
+	 * Indicates if the type is considered a collection of elements (ie is an <code>Iterable</code>, an array or
+	 * a <code>Map</code>).
 	 *
-	 * @param type the type to inspect.
-	 *
-	 * @return Returns true if the type is indexable.
+	 * @param type the type to check.
+	 * @return returns <code>true</code> if <code>type</code> can be considered a collection of elements
 	 */
-	public static boolean isIndexable(Type type) {
-		boolean isIndexable = false;
-		if ( ReflectionHelper.isList( type ) ) {
-			isIndexable = true;
-		}
-		else if ( ReflectionHelper.isMap( type ) ) {
-			isIndexable = true;
-		}
-		else if ( TypeHelper.isArray( type ) ) {
-			isIndexable = true;
-		}
-		return isIndexable;
+	public static boolean isElementCollection(Type type) {
```

Comment:
I'd prefer `isCollection()` or `isCollectionType()`.

"element collection" reminds too much of the construct with the same name in JPA (`@ElementCollection`). Here I'd refer to any collection-kind type by "collection" including maps, i.e. not `java.util.Collection`.